controllers/imageclusterinstall_controller.go

```diff
@@ -723,7 +728,41 @@ func (r *ImageClusterInstallReconciler) imageSetRegistry(ctx context.Context, ic
 	return strings.Split(namedRef.Name(), "/")[0], nil
 }
 
+func (r *ImageClusterInstallReconciler) nmstateConfigFromBMH(ctx context.Context, ici *v1alpha1.ImageClusterInstall) (string, error) {
+	bmh, err := r.getBMH(ctx, ici.Spec.BareMetalHostRef)
```

Comment:
Can we pass the BMH through to this function instead of querying it multiple times?

Reply:
Actually as client has cache get command should cost nothing but code will be much cleaner, WDYT?